File: C-Solutions/Problem-03.c
/* What is the largest prime factor of the number 600851475143? */

#include <stdio.h>
#include <time.h>

int main() {
  clock_t begin = clock();

  printf("Hello World!\n");
  
  clock_t end = clock();
  double time_spent = (double)(end - begin) / CLOCKS_PER_SEC;
  printf("RunTime: %f\n", time_spent);
  return 0;
}

File: Julia-Solutions/Problem-03.jl
# What is the largest prime factor of the number 600851475143?

function isPrime(num)
  if num < 2
    return false
  elseif num == 2
    return true
  else
    return sum([1 for i in [2:Int(round(sqrt(num)));] if num % i == 0]) == 0
  end
end

function primeFactors(num)
  return [i for i in [2:Int(round(sqrt(num)));] if num % i == 0 && isPrime(i)]
end

primeFactors(600851475143)[end]
